### PR TITLE
Generate itemtypes without numbers; fixes #305

### LINKED ISSUE #305: Undefined index on dropdown field search list
Bonjour,

lorsque l'on créer un champ de type liste déroulante avec un chiffre par exemple poste8, 
la recherche dans les intitulés de GLPI ne fonctionne pas
 
Création du bloc et de champ :
image

Voici le résultat dans la liste des intitulés 
image

### PR DESCRIPTION
**Pull request by cedric-anne**
See https://github.com/glpi-project/glpi/pull/5495

### CODE REVIEW

File: inc/container.class.php [hunk 1]
```diff
@@ -68,6 +68,44 @@ static function install(Migration $migration, $version) {
...
       }
 
+      // Fix containers names that were generated prior to Fields 1.9.2.
+      $glpi_version = preg_replace('/^((\d+\.?)+).*$/', '$1', GLPI_VERSION);
+      $bad_named_containers = $DB->request(
+         [
+            'FROM' => self::getTable(),
+            'WHERE' => [
+               // Regex will be escaped by PDO in GLPI 10+, but has to be escaped for GLPI < 10
+               'name' => ['REGEXP', version_compare($glpi_version, '10.0', '>=') ? '\d+' : '\\\d+'],
```
Comment (trasher) [inline inc/container.class.php]: Maybe better to use escaping method rather than hardcode it

File: inc/toolbox.class.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    * @param string $label
+    *
+    * @return string
+    */
+   public function getSystemNameFromLabel($label) {
+
+      $name = strtolower($label);
+
+      // 1. remove trailing "s" (plural forms)
+      $name = preg_replace('/s*$/', '', $name);
```
Comment (trasher) [inline inc/toolbox.class.php]: There are methods in core to manage plural/singular

File: inc/container.class.php (hunk 1 above)
Comment (cedric-anne) [reply inc/container.class.php]: Done.

File: inc/toolbox.class.php (hunk 2 above)
Comment (cedric-anne) [reply inc/toolbox.class.php]: Right, thanks.

**Comment by cedric-anne** [comment]
@stonebuzz

I wait for your confirmation that problem is solved and migration is working well before merging this.

**Comment by cedric-anne** [comment]
Fix confirmed by @stonebuzz .

**Pull request merged**